### DISCUSSION TITLE
error: unknown type name 'uint'

### DISCUSSION THREAD

**Discussion by KarurNaveen** [babelouest/ulfius#215] [General]
Hi , 

when we integrated ulfius into our project and compile , we are facing the "unknown type uint" error.

/usr/local/include/ulfius.h:1484:3: error: unknown type name 'uint'
 1484 |   uint server_no_context_takeover;
      |   ^~~~
/usr/local/include/ulfius.h:1485:3: error: unknown type name 'uint'
 1485 |   uint client_no_context_takeover;
      |   ^~~~
/usr/local/include/ulfius.h:1486:3: error: unknown type name 'uint'
 1486 |   uint server_max_window_bits;
      |   ^~~~
/usr/local/include/ulfius.h:1487:3: error: unknown type name 'uint'
 1487 |   uint client_max_window_bits;
      |   ^~~~

struct _websocket_deflate_context {
  z_stream infstream;
  z_stream defstream;
  int deflate_mask;
  int inflate_mask;
  **uint server_no_context_takeover;
  uint client_no_context_takeover;
  uint server_max_window_bits;
  uint client_max_window_bits;**
};

Please do the needful , after changing to unsigned int errors are not seen.

Thanks
Naveen

**Comment by babelouest** [comment]
Hello,

This should more have been an issue rather than a discussion I guess...

Can you tell me what system you are using to compile ulfius?

**Reply by KarurNaveen** [reply]
Hi Nicolas , 

[root@clab1012node01 working_dir]# **cat /etc/redhat-release** 
Red Hat Enterprise Linux Server release 7.6 (Maipo)
[root@clab1012node01 working_dir]# **uname -a**
Linux clab1012node01 3.10.0-1160.6.1.el7.x86_64 #1 SMP Wed Oct 21 13:44:38 EDT 2020 x86_64 x86_64 x86_64 GNU/Linux

Thanks
Naveen

**Comment by babelouest** [comment]
Hello @KarurNaveen ,

If you replace the `uint` with `unsigned int`, does it correctly compile? If so, does you program or the example programs work correctly as well?

**Reply by KarurNaveen** [reply]
Hi Nicolas ,

Yes if i changed to unsigned int , it will work fine and all the example programs works fine.

Thanks
Naveen

**Comment by babelouest** [comment]
I've made the changes in this commit https://github.com/babelouest/ulfius/commit/618d67209b17ce1bd75c4db00450f4b93ca3cbed , should be good now, let me know @KarurNaveen if it's all good
